apm-agent-plugins/apm-api-plugin/src/main/java/co/elastic/apm/agent/pluginapi/ElasticApmApiInstrumentation.java

```diff
@@ -147,4 +148,16 @@ public static void captureException(@Advice.Origin Class<?> clazz, @Advice.Argum
         }
     }
 
+    public static class SetServiceInfoForClassLoaderInstrumentation extends ElasticApmApiInstrumentation {
+        public SetServiceInfoForClassLoaderInstrumentation() {
+            super(named("setServiceInfoForClassLoader"));
+        }
+
+        public static class AdviceClass {
+            @Advice.OnMethodExit(suppress = Throwable.class, inline = false)
+            public static void setServiceInfoForClassLoader(@Advice.Argument(0) @Nullable ClassLoader classLoader, @Advice.Argument(1) String serviceName, @Advice.Argument(2) @Nullable String serviceVersion) {
+                tracer.setServiceInfoForClassLoader(classLoader, ServiceInfo.of(serviceName, serviceVersion));
```

Comment:
I wonder if we should check for `serviceName` being `null`

Reply:
A `null` service name will have no effect here - it won't be mapped and not even cause allocation of a `ServiceInfo` object. 
However, I wonder if we should allow `null` service version going forward, now that #1726 is merged.
I think providing a service name without a version should be considered a bug, because it means we will use the global service version for a manually set service name, and there's no real reason why those would be related.
@felixbarny WDYT? Should we only allow overriding these two values together?

Reply:
I think its is fine to set a service name without a version. The global service version won't be used in such a case. But there service version will be serialized as null and dropped when ingested by the apm-server

Reply:
You mean - currently APM Server ignores the global version when we use a service name in transaction/metric context?

Reply:
Sorry, my bad. This logic is only implemented for metrics. I created elastic/apm-server#7281 to fix this for transactions also.